devito/types/basic.py

```diff
@@ -222,6 +222,10 @@ def _mem_shared_remote(self):
         """
         return False
 
+    @property
```

Comment:
I would make this an attribute of the class rather than the object. I.e. have `_restrict_keyword = 'restrict'` immediately under the docstring.